File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Effect/Combat/IntenseZoomEffect.ts
import Color from "../../DataDefine/Color";
import DungeonManager from "../../DungeonManager";
import { AbstractEffect } from "../AbstractEffect";
import { BorderFlashEffect } from "./BorderFlashEffect";


export class IntenseZoomEffect extends AbstractEffect{
    public constructor(private _X:number,private _Y:number,private _IsBlack:boolean) {
        super();
    }
    Update() {
        super.Update();
        DungeonManager.EffectManager.Play(new BorderFlashEffect(this._IsBlack ? Color.Black : Color.Gold,this._IsBlack));
        for (let i = 0; i < 10; i++) {
            DungeonManager.EffectManager.Play(new IntenseZoomEffect(this._X,this._Y,this._IsBlack));
        }
        this.IsDone = true;
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Combat/IntenseZoomEffect.cs
using ModTheSpire.DataDefine;
using ModTheSpire.Dungeon;
using UnityEngine;

namespace ModTheSpire.Effect.Combat
{
    public class IntenseZoomEffect :AbstractEffect
    {
        private bool isBlack;
        private float x;
        private float y;
        private static int AMT = 10;

        public IntenseZoomEffect(float x, float y, bool isBlack) {
            this.x = x;
            this.y = y;
            this.isBlack = isBlack;
        }

        public override void Update()
        {
            base.Update();
            if (this.isBlack) {
                DungeonManager.EffectManager.Play(new BorderFlashEffect(Color.black, this.isBlack));
            } else {
                DungeonManager.EffectManager.Play(new BorderFlashEffect(MTSColor.Gold, this.isBlack));
            }
            for (int i = 0; i < 10; ++i) {
                DungeonManager.EffectManager.Play(new IntenseZoomParticle(this.x, this.y, this.isBlack));
            }
            Finish();
        }
    }
}